src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -75,6 +79,38 @@
    (clear-both)])
 
 
+(react/defc MethodDetailsViewer
+  {:render
+   (fn [{:keys [props state]}]
+     [:div {:style {:marginRight "20%"}}
+      (cond
+        (:loaded-method @state)
+        [:div {}
+         [comps/EntityDetails
+          {:entity (:loaded-method @state)}]]
+        :else (render-top-bar-config (:config props)))])
+   :component-did-mount
+   (fn [{:keys [props state]}]
+     (endpoints/call-ajax-orch
+       {:endpoint (endpoints/get-agora-method
+                    (:namespace props)
+                    (:name props)
+                    (:snapshotId props))
+        :headers {"Content-Type" "application/json"}
+        :on-done (fn [{:keys [success? get-parsed-response status-text]}]
+                   (if success?
+                     (swap! state assoc :loaded-method (get-parsed-response))
+                     (swap! state assoc :error status-text)))}))})
+
+(defn- render-top-bar [config]
```

Comment:
The relationship between this and `render-top-bar-config` is not very clear to me. Can you do some renaming to add clarity?